eZ/Bundle/EzPublishIOBundle/Command/MigrateFilesCommand.php

```
@@ -0,0 +1,312 @@
+<?php
+
+/**
+ * File containing the MigrateFilesCommand class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishIOBundle\Command;
+
+use eZ\Bundle\EzPublishIOBundle\Migration\FileListerRegistry;
+use eZ\Bundle\EzPublishIOBundle\Migration\FileMigratorInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+final class MigrateFilesCommand extends Command
+{
+    /** @var mixed Configuration for metadata handlers */
+    private $configuredMetadataHandlers;
+
+    /** @var mixed Configuration for binary data handlers */
+    private $configuredBinarydataHandlers;
+
+    /** @var \eZ\Bundle\EzPublishIOBundle\Migration\FileListerRegistry */
+    private $fileListerRegistry;
+
+    /** @var \eZ\Bundle\EzPublishIOBundle\Migration\FileListerInterface[] */
+    private $fileListers;
+
+    /** @var \eZ\Bundle\EzPublishIOBundle\Migration\FileMigratorInterface */
+    private $fileMigrator;
+
+    public function __construct(
+        $configuredMetadataHandlers,
+        $configuredBinarydataHandlers,
+        FileListerRegistry $fileListerRegistry,
+        FileMigratorInterface $fileMigrator
+    ) {
+        $this->configuredMetadataHandlers = $configuredMetadataHandlers;
+        $this->configuredBinarydataHandlers = $configuredBinarydataHandlers;
+        $this->configuredMetadataHandlers['default'] = [];
```

@bdunogier They are silently ignored. I could ignore them more loudly (log warning or even fail).